tests/data/test_dataset.py

```diff
@@ -12,386 +12,378 @@
+        """Test that read() can load attributes from a Dataset file."""
+
+        existing = Dataset()
+
+        existing.x = 1
+        existing.y = attribute("abc", doc="abc")
+        existing.z = [1, 2, 3]
+
+        dset = Dataset(x=2)
+        dset.read(existing, attributes=attributes_arg)
+
+        assert dset.list_attributes() == attributes_expect
+        assert dset.y == "abc"
+        assert dset.attr_info["y"].doc == "abc"
+
+    @pytest.mark.parametrize("overwrite, expect_x", [(False, 2), (True, 1)])
+    def test_read_overwrite(self, overwrite, expect_x):
+        """Test that overwrite determnines whether an existing attribute
+        is overwritten by read()."""
+
+        existing = Dataset(x=1)
+        new = Dataset(x=2)
+
+        new.read(existing, overwrite=overwrite)
+        assert new.x == expect_x
+
+    @pytest.mark.parametrize("mode", ["w-", "w", "a"])
+    def test_write(self, tmp_path, mode):
+        """Test that the ``write`` method creates a Zarr file that contains
+        the all the data in the dataset."""
+        ds = Dataset(x=DatasetScalar(1.0, AttributeInfo(py_type=int, doc="an int")))
+
+        path: Path = tmp_path / "test"
+        ds.write(path, mode=mode)
+
+        zgrp = open_group(path, mode="r")
+
+        ds_2 = Dataset(zgrp)
+
+        assert ds_2.bind is not ds.bind
+        assert ds.attrs == ds_2.attrs
+
+    def test_init_subclass(self):
+        """Test that __init_subclass__() does the following:
+
+        - Does not create fields for _InitArg and ClassVar
+        - collects fields from type annotations, including py_type
+        - gets data name and params from class arguments
+        """
+
+        class NewDataset(
+            Dataset, data_name="new_dataset", params=("x", "y")
+        ):  # pylint: disable= too-few-public-methods
+            """Dataset"""
+
+            class_info: ClassVar[str] = "Class variable"
+
+            init_arg: int = _init_arg(1)
+
+            x: int
+            y: str = field(doc="y documentation")
+            jsonable: Dict[str, str] = field(DatasetJSON, doc="json data")
+
+        ds = NewDataset(init_arg=3, x=1, y="abc", jsonable={"a": "b"})
+
+        assert ds.data_name == "new_dataset"
+        assert ds.params == {"x": 1, "y": "abc"}
+
+        assert ds.attr_info["x"].py_type == "int"
+        assert ds.attr_info["y"].py_type == "str"
+        assert ds.attr_info["y"].doc == "y documentation"
+        assert ds.attr_info["jsonable"].py_type == "dict[str, str]"
+        assert ds.attr_info["jsonable"].doc == "json data"
 
+    def test_dataset_as_attribute(self):
+        """Test that a Dataset can be an attribute of
+        another dataset."""
 
-def test_repr_non_standard():
-    """Test that __repr__ for non-standard Datasets look as expected."""
-    dataset = qml.data.Dataset(foo=1, bar=2)
-    assert repr(dataset) == "<Dataset = attributes: ['foo', 'bar']>"
+        child = Dataset(x=1, y=2)
```

Comment:
These can be unindented to one level.

Reply:
I'm not sure what you mean - the whole test class?